<a href="https://colab.research.google.com/github/skdarkey/Selorm_Portfolio/blob/main/Google_Play_Store_App_Analytics.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Introduction

In this notebook, I did a comprehensive analysis of the Android app market by comparing thousands of apps in the Google Play store.


* The goal is to understand the android application store to enable me price my future applications reasonably. My colleague developers can also make use of this analytics to place the apps in the appropriate categories and adopt best pricing mechanisms backed by data to ensure success of their apps.

# About the Dataset of Google Play Store Apps & Reviews

**Data Source:** <br>
The original App and review data was scraped from the Google Play Store by Lavanya Gupta in 2018. Original data can be found here [here](
https://www.kaggle.com/lava18/google-play-store-apps) on Kaggle.

# Import Statements

In [ ]:
import pandas as pd
import plotly.express as px

# Notebook Presentation

In [ ]:
# Show numeric output in decimal format e.g., 2.15
pd.options.display.float_format = '{:,.2f}'.format

# Read the Dataset

In [ ]:
data = "https://raw.githubusercontent.com/skdarkey/Datasets/main/Google_Play_Store_Apps_Analytics.csv"
#df_apps = pd.read_csv('apps.csv')
df_apps = pd.read_csv(data)

# Data Cleaning

Knowing the number of rows and columns the `df_apps` have? What the column names are?

In [ ]:
df_apps.shape

(10841, 12)

In [ ]:
# Getting the column names in the data set
df_apps.columns

Index(['App', 'Category', 'Rating', 'Reviews', 'Size_MBs', 'Installs', 'Type',
       'Price', 'Content_Rating', 'Genres', 'Last_Updated', 'Android_Ver'],
      dtype='object')

In [ ]:
# Inspecting random data in the data set
df_apps.sample(5)

App       Category  Rating  Reviews  Size_MBs     Installs  \
805     Prestige AH        MEDICAL     4.0        2     29.00          100   
1415     CX Capture       BUSINESS     NaN        1      5.80          100   
1593         DG-App          TOOLS     5.0        1      4.50          500   
457    FN Web Radio  COMMUNICATION     NaN        0      1.60           10   
10381    Google Pay        FINANCE     4.2   347838      9.15  100,000,000   

       Type Price Content_Rating         Genres       Last_Updated  \
805    Free     0       Everyone        Medical       May 11, 2017   
1415   Free     0       Everyone       Business   January 23, 2018   
1593   Free     0       Everyone          Tools  November 29, 2017   
457    Free     0       Everyone  Communication       May 18, 2018   
10381  Free     0       Everyone        Finance      July 26, 2018   

              Android_Ver  
805          4.0.3 and up  
1415         4.0.3 and up  
1593           4.1 and up  
457            2.3 and up  
10381  Varies with device

### Droping some Unused Columns

** `Last_Updated` and `Android_Version` from the DataFrame will not be used.

In [ ]:
df_apps.drop(['Last_Updated', 'Android_Ver'], axis=1, inplace=True)

### Finding and Removing NaN values in Ratings column


In [ ]:
# getting to know number of columns containing 'Not-a-Number' values in the Dataset.
df_apps.Rating.isna().values.sum()

1474

In [ ]:
#droping columns with NaN Values in the data set
df_apps_clean = df_apps.dropna()
print(df_apps_clean.head())
df_apps_clean.isna().values.any()

                                                App Category  Rating  Reviews  \
21                              KBA-EZ Health Guide  MEDICAL    5.00        4   
28                                         Ra Ga Ba     GAME    5.00        2   
47                                          Mu.F.O.     GAME    5.00        2   
82                                 Brick Breaker BR     GAME    5.00        7   
99  Anatomy & Physiology Vocabulary Exam Review App  MEDICAL    5.00        1   

    Size_MBs Installs  Type  Price Content_Rating   Genres  
21     25.00        1  Free      0       Everyone  Medical  
28     20.00        1  Paid  $1.49       Everyone   Arcade  
47     16.00        1  Paid  $0.99       Everyone   Arcade  
82     19.00        5  Free      0       Everyone   Arcade  
99      4.60        5  Free      0       Everyone  Medical  


False

### Finding and Removing Duplicates



In [ ]:
#Checking for duplicates in the data
df_apps_clean.duplicated().values.any()                 #Are there any duplicates in the data
df_apps_clean.duplicated().values.sum()                 #In total, how many duplicated rows do we have

476

In [ ]:
# providing specific column names to be used to check and remove duplicates
df_apps_clean = df_apps_clean.drop_duplicates(subset=['App', 'Type', 'Price'])


In [ ]:
# Checking for a specific name with duplicates.
df_apps_clean[df_apps_clean.App == 'Instagram']

App Category  Rating   Reviews  Size_MBs       Installs  Type  \
10806  Instagram   SOCIAL    4.50  66577313      5.30  1,000,000,000  Free   

      Price Content_Rating  Genres  
10806     0           Teen  Social

In [ ]:
df_apps_clean.shape
df_apps_clean.describe()  # getting descriptive stats

Rating       Reviews  Size_MBs
count 8,199.00      8,199.00  8,199.00
mean      4.17    255,072.97     20.09
std       0.54  1,985,241.40     21.65
min       1.00          1.00      0.01
25%       4.00        126.00      4.90
50%       4.30      2,992.00     11.00
75%       4.50     43,612.50     28.00
max       5.00 78,158,306.00    100.00

# Finding the Highest Rated Apps


In [ ]:
# Identifying highest rated Apps
High_Rated = df_apps_clean.sort_values('Rating', ascending=False)

# Identifying highest rated Apps that also have highest reviews
High_Rated_AndReviewed = High_Rated.sort_values('Reviews', ascending=False)

#Print samples of each to compare
print(High_Rated.head())
print(High_Rated_AndReviewed.head())

                      App     Category  Rating  Reviews  Size_MBs Installs  \
21    KBA-EZ Health Guide      MEDICAL    5.00        4     25.00        1   
1230         Sway Medical      MEDICAL    5.00        3     22.00      100   
1227    AJ Men's Grooming    LIFESTYLE    5.00        2     22.00      100   
1224       FK Dedinje BGD       SPORTS    5.00       36      2.60      100   
1223      CB VIDEO VISION  PHOTOGRAPHY    5.00       13      2.60      100   

      Type Price Content_Rating       Genres  
21    Free     0       Everyone      Medical  
1230  Free     0       Everyone      Medical  
1227  Free     0       Everyone    Lifestyle  
1224  Free     0       Everyone       Sports  
1223  Free     0       Everyone  Photography  
                                            App       Category  Rating  \
10805                                  Facebook         SOCIAL    4.10   
10785                        WhatsApp Messenger  COMMUNICATION    4.40   
10806                      

# Find 5 Largest Apps in terms of Size (MBs)


In [ ]:
LargestApps = df_apps_clean.sort_values('Size_MBs', ascending=False)
LargestApps[:5]

App            Category  Rating  Reviews  \
9942   Talking Babsy Baby: Baby Games           LIFESTYLE    4.00   140995   
10687          Hungry Shark Evolution                GAME    4.50  6074334   
9943            Miami crime simulator                GAME    4.00   254518   
9944     Gangster Town: Vice District              FAMILY    4.30    65146   
3144                       Vi Trainer  HEALTH_AND_FITNESS    3.60      124   

       Size_MBs     Installs  Type Price Content_Rating  \
9942     100.00   10,000,000  Free     0       Everyone   
10687    100.00  100,000,000  Free     0           Teen   
9943     100.00   10,000,000  Free     0     Mature 17+   
9944     100.00   10,000,000  Free     0     Mature 17+   
3144     100.00        5,000  Free     0       Everyone   

                       Genres  
9942   Lifestyle;Pretend Play  
10687                  Arcade  
9943                   Action  
9944               Simulation  
3144         Health & Fitness


*Based on the data, there seems to be an upper limit in size for apps that can be placed on the Google App store.*

# Finding the top 50 Apps with Most Reviews


In [ ]:
# Identifying highest reviewed Apps
Highest_Reviewed = df_apps_clean.sort_values('Reviews', ascending=False)
HighR50 = Highest_Reviewed[:50]
print(HighR50)

# Are there any paid applications in the top 50 most reviewed apps
HighR50[HighR50.Type == 'Paid'].values.any()

                                                     App            Category  \
10805                                           Facebook              SOCIAL   
10785                                 WhatsApp Messenger       COMMUNICATION   
10806                                          Instagram              SOCIAL   
10784           Messenger – Text and Video Chat for Free       COMMUNICATION   
10650                                     Clash of Clans                GAME   
10744            Clean Master- Space Cleaner & Antivirus               TOOLS   
10835                                     Subway Surfers                GAME   
10828                                            YouTube       VIDEO_PLAYERS   
10746  Security Master - Antivirus, VPN, AppLock, Boo...               TOOLS   
10584                                       Clash Royale                GAME   
10763                                   Candy Crush Saga                GAME   
10770        UC Browser - Fast Download 

False


*Facebook, WhatsApp, Messenger, Instagram have the highest reviews. I believe all these apps belong to one company 'Meta'. How? Interestingly too, there seems to be no paid app in the top 50 most reviewed apps in the app store.*

# Visualising Apps' Content Ratings using Plotly Pie and Donut Charts -



In [ ]:
# Counting the number of Rating occurences for each content-Type category of the Apps
ratings = df_apps_clean.Content_Rating.value_counts()
ratings
# df_apps_clean[df_apps_clean.Content_Rating == 'Adults only 18+']

Everyone           6621
Teen                912
Mature 17+          357
Everyone 10+        305
Adults only 18+       3
Unrated               1
Name: Content_Rating, dtype: int64

*It seems Adults only apps are so few. May be adults are not very tech savvy. Apps with contents targeted at everyone and teens seem so popular.*

In [ ]:
# Creating a chart to display the content rating and percentage of apps in each content category
fig = px.pie(labels=ratings.index, values=ratings.values,
             title='Content Rating', names=ratings.index)

fig.update_traces(textposition='outside', textinfo='percent+label')

fig.show()

/usr/local/lib/python3.10/dist-packages/plotly/express/_core.py:137: FutureWarning: Support for multi-dimensional indexing (e.g. `obj[:, None]`) is deprecated and will be removed in a future version.  Convert to a numpy array before indexing instead.
  return args["labels"][column]


In [ ]:
# How about the same data on doughnut chart
fig = px.pie(labels=ratings.index, values=ratings.values,
             title='Content Rating', names=ratings.index,
             hole = 0.6)  # hole for doughnut

fig.update_traces(textposition='inside', textfont_size = 15, textinfo='percent')

fig.show()

/usr/local/lib/python3.10/dist-packages/plotly/express/_core.py:137: FutureWarning:

Support for multi-dimensional indexing (e.g. `obj[:, None]`) is deprecated and will be removed in a future version.  Convert to a numpy array before indexing instead.



# Examining the Number of Installs

Here I want to know the number of installs per app.

*Answering the question how many apps had several millions of installations? vs How many apps just had a single install?*

In [ ]:
# Using the describe method or the info method to know the data type of the install column
df_apps_clean.Installs.describe()
df_apps_clean.info()

#Using the group and count method to know the number of Apps installed at each level. Egs; Apps that have only 1 install, or over 1000 installs
df_apps_clean[['App', 'Installs']].groupby('Installs').count()

#High_install = df_apps_clean.sort_values('Installs', ascending=False)
#High_install.head()


<class 'pandas.core.frame.DataFrame'>
Int64Index: 8199 entries, 21 to 10835
Data columns (total 10 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   App             8199 non-null   object 
 1   Category        8199 non-null   object 
 2   Rating          8199 non-null   float64
 3   Reviews         8199 non-null   int64  
 4   Size_MBs        8199 non-null   float64
 5   Installs        8199 non-null   object 
 6   Type            8199 non-null   object 
 7   Price           8199 non-null   object 
 8   Content_Rating  8199 non-null   object 
 9   Genres          8199 non-null   object 
dtypes: float64(2), int64(1), object(7)
memory usage: 704.6+ KB


App
Installs           
1                 3
1,000           698
1,000,000      1417
1,000,000,000    20
10               69
10,000          988
10,000,000      933
100             303
100,000        1096
100,000,000     189
5                 9
5,000           425
5,000,000       607
50               56
50,000          457
50,000,000      202
500             199
500,000         504
500,000,000      24

In [ ]:
# Removing the comma in the figures above.
df_apps_clean.Installs = df_apps_clean.Installs.astype(str).str.replace(',', "")
df_apps_clean.Installs.sample(5)

7933     1000000
9496    10000000
2460        1000
4076       10000
1319         100
Name: Installs, dtype: object

In [ ]:
#Converting the new figures now to numeric data types
df_apps_clean.Installs = pd.to_numeric(df_apps_clean.Installs)

In [ ]:
df_apps_clean[['App', 'Installs']].groupby('Installs').count()


App
Installs        
1              3
5              9
10            69
50            56
100          303
500          199
1000         698
5000         425
10000        988
50000        457
100000      1096
500000       504
1000000     1417
5000000      607
10000000     933
50000000     202
100000000    189
500000000     24
1000000000    20

### Finding the Most Expensive Apps

* The approach here is to filter out any Junk, and calculating a Sales Revenue Estimate.


* Converting the price column to numeric data. Then investigate the top 20 most expensive apps in the dataset.



In [ ]:
# Converting the price column to numeric
df_apps_clean.Price = pd.to_numeric(df_apps_clean.Price.str.replace('$', ''))

df_apps_clean.Price.head()

<ipython-input-27-b496e0f105a6>:2: FutureWarning:

The default value of regex will change from True to False in a future version. In addition, single character regular expressions will *not* be treated as literal strings when regex=True.



21   0.00
28   1.49
47   0.99
82   0.00
99   0.00
Name: Price, dtype: float64

### The most expensive apps

In [ ]:
# Sorting the apps by price to find the most expensive
PriceSorted = df_apps_clean.sort_values('Price', ascending=False)
PriceSorted[:20]

# removing some expensive replica apps
df_apps_clean = df_apps_clean[df_apps_clean['Price'] < 250]

df_apps_clean.sort_values('Price', ascending=False).head()


App   Category  Rating  Reviews  Size_MBs  \
2281  Vargo Anesthesia Mega App    MEDICAL    4.60       92     32.00   
1407               LTC AS Legal    MEDICAL    4.00        6      1.30   
2629           I am Rich Person  LIFESTYLE    4.20      134      1.80   
2481    A Manual of Acupuncture    MEDICAL    3.50      214     68.00   
2463         PTA Content Master    MEDICAL    4.20       64     41.00   

      Installs  Type  Price Content_Rating     Genres  
2281      1000  Paid  79.99       Everyone    Medical  
1407       100  Paid  39.99       Everyone    Medical  
2629      1000  Paid  37.99       Everyone  Lifestyle  
2481      1000  Paid  33.99       Everyone    Medical  
2463      1000  Paid  29.99       Everyone    Medical

### Estimating Highest Grossing Paid Apps

In [ ]:
# Adding a new column and calculating the estimated revenue for the apps
df_apps_clean['Revenue_Estimate'] = df_apps_clean.Installs.mul(df_apps_clean.Price)

# Sorting the Revenue Estimates to see which apps have gotten the most revenues
df_apps_clean.sort_values('Revenue_Estimate', ascending=False)[:10]

App     Category  Rating  Reviews  Size_MBs  \
9220                      Minecraft       FAMILY    4.50  2376564     19.00   
8825                  Hitman Sniper         GAME    4.60   408292     29.00   
7151  Grand Theft Auto: San Andreas         GAME    4.40   348962     26.00   
7477            Facetune - For Free  PHOTOGRAPHY    4.40    49553     48.00   
7977        Sleep as Android Unlock    LIFESTYLE    4.50    23966      0.85   
6594            DraStic DS Emulator         GAME    4.60    87766     12.00   
6082                   Weather Live      WEATHER    4.50    76593      4.75   
7954                    Bloons TD 5       FAMILY    4.60   190086     94.00   
7633        Five Nights at Freddy's         GAME    4.60   100805     50.00   
6746     Card Wars - Adventure Time       FAMILY    4.30   129603     23.00   

      Installs  Type  Price Content_Rating                     Genres  \
9220  10000000  Paid   6.99   Everyone 10+  Arcade;Action & Adventure   
8825  10000000  Paid   0.99     Mature 17+                     Action   
7151   1000000  Paid   6.99     Mature 17+                     Action   
7477   1000000  Paid   5.99       Everyone                Photography   
7977   1000000  Paid   5.99       Everyone                  Lifestyle   
6594   1000000  Paid   4.99       Everyone                     Action   
6082    500000  Paid   5.99       Everyone                    Weather   
7954   1000000  Paid   2.99       Everyone                   Strategy   
7633   1000000  Paid   2.99           Teen                     Action   
6746   1000000  Paid   2.99   Everyone 10+    Card;Action & Adventure   

      Revenue_Estimate  
9220     69,900,000.00  
8825      9,900,000.00  
7151      6,990,000.00  
7477      5,990,000.00  
7977      5,990,000.00  
6594      4,990,000.00  
6082      2,995,000.00  
7954      2,990,000.00  
7633      2,990,000.00  
6746      2,990,000.00

# Analysing App Categories using Plotly Bar Charts & Scatter Plots

In [ ]:
#Finding the number of unique categories in the data.
df_apps_clean.Category.nunique()

33

In [ ]:
# Number of apps by category
top10_category = df_apps_clean.Category.value_counts()[:10]

In [ ]:
top10_category

FAMILY             1606
GAME                910
TOOLS               719
PRODUCTIVITY        301
PERSONALIZATION     298
LIFESTYLE           297
FINANCE             296
MEDICAL             292
PHOTOGRAPHY         263
BUSINESS            262
Name: Category, dtype: int64

### Vertical Bar Chart
* Visualising categories with the highest number of Apps.

In [ ]:
# Ploting the graphs using plotly express
bar = px.bar(x = top10_category.index,
             y = top10_category.values)
bar.show()

*It can be seen that Family category has the highest number of Apps*

### Most Popular Categories (highest Downloads)
* By Horizontal Bar Chart

In [ ]:
# grouping the apps by category and sum the number of installs per category to give idea on most popular categories
category_installs = df_apps_clean.groupby('Category').agg({'Installs': pd.Series.sum})
category_installs.sort_values('Installs', ascending=True, inplace=True)

In [ ]:
#Plotting the bars horizontally
h_bar = px.bar(x = category_installs.Installs,
               y = category_installs.index,
               orientation='h')
h_bar.show()


*It is seen now that the GAMES category has the highest number of downloads and hence popular even more than the Family category that has about 2 times the number of games apps*

### Category Concentration - Downloads vs. Competition

* The goal is to know the number of apps per category and total installs of those apps

In [ ]:
#Creating a new dataframe
CategoryNumber = df_apps_clean.groupby('Category').agg({'App': pd.Series.count})

#Using .merge() I joined the 2 dataframes
cat_merged_df = pd.merge(CategoryNumber, category_installs, on='Category', how="inner")

#Checking the new number of row and columns
print(f'The dimensions of the DataFrame are: {cat_merged_df.shape}')

#Sorting
cat_merged_df.sort_values('Installs', ascending=False)

The dimensions of the DataFrame are: (33, 2)


App     Installs
Category                              
GAME                  910  13858762717
COMMUNICATION         257  11039241530
TOOLS                 719   8099724500
PRODUCTIVITY          301   5788070180
SOCIAL                203   5487841475
PHOTOGRAPHY           263   4649143130
FAMILY               1606   4437554490
VIDEO_PLAYERS         148   3916897200
TRAVEL_AND_LOCAL      187   2894859300
NEWS_AND_MAGAZINES    204   2369110650
ENTERTAINMENT         102   2113660000
BOOKS_AND_REFERENCE   169   1665791655
PERSONALIZATION       298   1532352930
SHOPPING              180   1400331540
HEALTH_AND_FITNESS    243   1134006220
SPORTS                260   1096431465
BUSINESS              262    692018120
LIFESTYLE             297    503611120
MAPS_AND_NAVIGATION   118    503267560
FINANCE               296    455249400
WEATHER                72    361096500
EDUCATION             118    352852000
FOOD_AND_DRINK         94    211677750
DATING                134    140912410
ART_AND_DESIGN         61    114233100
HOUSE_AND_HOME         62     97082000
AUTO_AND_VEHICLES      73     53129800
LIBRARIES_AND_DEMO     64     52083000
COMICS                 54     44931100
MEDICAL               292     39162676
PARENTING              50     31116110
BEAUTY                 42     26916200
EVENTS                 45     15949410

In [ ]:
#Now I plot the new new dataframe as scatter plot to measure Category concentration of installed applications

scatter = px.scatter(cat_merged_df,
                     x = 'App',
                     y = 'Installs',
                     title = 'Category Concentration',
                     size = 'App',
                     hover_name= cat_merged_df.index,
                     color = 'Installs')
scatter.update_layout(xaxis_title = "Number of Apps in Category (Lower = More Concentration)",
                      yaxis_title = "Installs",
                      yaxis=dict(type = 'log'))
scatter.show()


*The chart here is showing that categories like Family, Tools and Game have many different apps sharing the high number of downloads. But for categories like video players and entertainment (upper right), only few apps dominate the downloads.The bottom left categories do not have high apps nor high downloads. So increasing your new application download rate, it is wise to place it in the Family, Games and Tools category.*

### Extracting Nested Data from a Column



In [ ]:
# No of unique Genres
df_apps_clean.Genres.nunique()
# Getting a sample of them to inspect
df_apps_clean.Genres.sample(10)
# Counting the No of genres in categories. Nested values are present
df_apps_clean.Genres.value_counts()

Tools                                718
Entertainment                        467
Education                            429
Productivity                         301
Personalization                      298
                                    ... 
Adventure;Brain Games                  1
Travel & Local;Action & Adventure      1
Art & Design;Pretend Play              1
Music & Audio;Music & Video            1
Lifestyle;Pretend Play                 1
Name: Genres, Length: 114, dtype: int64

In [ ]:
# Separating the nested genre names by Spliting the strings and then stack them
stack = df_apps_clean.Genres.str.split(';', expand=True).stack()
stack.head()

#printing a sample to inspect
print(f'I now have a single Genre column with shape: {stack.shape}')

# Count and Check split was successful
num_genres = stack.value_counts()
print(f'Number of genres: {len(num_genres)}')

I now have a single Genre column with shape: (8564,)
Number of genres: 53


# Examining the Competition in Genres
* Exploring also colour Scales in Plotly



In [ ]:
# plotting the genre data on vertical
Genre_Apps = df_apps_clean.groupby('Genres').agg({'App': pd.Series.count})
#Genre_Apps.head()

#plotting my charts with continuous color scale in plotly
v_bar = px.bar(
               x = num_genres.index[:15],           #index = number of category names
               y = num_genres.values[:15],
               title = 'Top Genres',
               hover_name= num_genres.index[:15],
               color = num_genres.values[:15],
               color_continuous_scale = 'Agsunset')
v_bar.update_layout(xaxis_title = "Genres",
               yaxis_title = "Number of Apps",
               coloraxis_showscale=False)

v_bar.show()

# Analysing Free vs. Paid Apps per Category
*  Exploring Grouped Bar Charts

In [ ]:
#Grouping the data by category, Type and aggregating the number of apps in the categories by type (Free of paid)
df_free_vs_paid = df_apps_clean.groupby(["Category", "Type"], as_index=False).agg({'App': pd.Series.count})
df_free_vs_paid.shape

(61, 3)

In [ ]:
# Plotling the free vs paid app per category and grouped bar charts

g_bar = px.bar(df_free_vs_paid,
               x = 'Category',
               y= 'App',
               title = 'Free vs Paid Apps by Category',
               color = 'Type',
               barmode = 'group')
g_bar.update_layout(xaxis_title = 'Category',
                    yaxis_title = 'Number of Apps',
                    xaxis = {'categoryorder': 'total descending'},
                    yaxis = dict(type = 'log'))
g_bar.show()

# Exploring Lost Downloads for Paid Apps
* Here the goal is to know how much downloads do paid application sacrifice. Thats comparing to being free apps.



In [ ]:
# And how many downloads are you giving up by making your app paid?

#Using the Box plots
box= px.box(df_apps_clean,
            y='Installs',
            x='Type',
            color='Type',
            notched=True,
            points = 'all',
            title = 'How Many Downloads are Paid Apps Giving Up?')
box.update_layout(yaxis=dict(type='log'))

box.show()

*Using the box plot, the median number of installations compare for a free app is 500k while that of paid app is 5k. The difference is large. Meaning if you choose to make your app paid, you are potentially giving up One hundred times rate of downloads.*

 *Things must be well thought out. The monetization strategies should decide such as ad revenue should be considered here. But, lets first of all know if paid apps make money.*

# Investigating Revenue by App Category.

In [ ]:
#How much revenue can we estimate per category
df_paid_apps = df_apps_clean[df_apps_clean['Type'] == 'Paid']

box = px.box(df_paid_apps,
             x = 'Category',
             y = 'Revenue_Estimate',
             title = "How much can paid Apps Earn?")
box.update_layout(xaxis_title = "Category",
                  yaxis_title = 'Paid App Ballpark Revenue',
                  xaxis = {'categoryorder': 'min ascending'},
                  yaxis = dict(type='log'))
box.show()

*Looking at the hover text, the median app in Tools category for example makes 5990 usd. Meaning many apps in the Tool category do not make much money, if you should take into consideration the amount spent in developing the app. However, the best apps in the same category can earn up to 3 Million dollars. Apps in Entertainment generally earn around 400K USD.*

*Hovering on the bars reveals a lot of earning potential informations for apps in each category.*


# Finally, examining Paid App Pricing Strategies by Category

* Meaning, how much can one charge for a new app if you want it paid.


In [ ]:
# App Pricing by Category
paid_app_median_price = df_paid_apps.Price.median()
print(f"The median price for a paid app is {paid_app_median_price}")

The median price for a paid app is 2.99


In [ ]:
# ploting price distribution for paid apps by category.
box = px.box(df_paid_apps,
             x = 'Category',
             y = "Price",
             title = 'Price per Category')

box.update_layout(xaxis_title = "Category",
                  yaxis_title = 'Paid App Price',
                  xaxis = {'categoryorder': 'max descending'},
                  yaxis = dict(type='log'))
box.show()

*The median price for a paid app is `$`2.99. The Dating Apps category has the highest average price of about `$`7 USD. The Social, Art_And_Design, and Auto_And_Vehicles category on the other hand have the lowest average price. This must be because there are a lot of free Social app, their revenue strategies is usually ad based. Medical apps also seem to be well patronised and have an average of `$`5.50. The spread seems wide for Maps and Navigation apps, with a median of 2.99 with a lower limit at 0.99 USD and a third quartile limit of 10.5 USD.*

That is a total wrap on the app market. Buy this analysis, if you want to launch your app next so you can be better placed to succeed 😎
